prog/alloc.go

```diff
@@ -51,13 +51,17 @@ func (ma *memAlloc) noteAlloc(addr0, size0 uint64) {
 	}
 }
 
-func (ma *memAlloc) alloc(r *randGen, size0 uint64) uint64 {
+func (ma *memAlloc) alloc(r *randGen, size0 uint64, alignment uint64) uint64 {
```

Comment:
I think you already went through the trouble of understanding what all this is doing.

If you don't mind, could you add high-level comments? I like summaries above the function describing what it does and the arguments. Optionally, if some things inline are non-trivial, inline comments might also be helpful.

Similar for all the other functions that you had to go through and understand.

[ Disclaimer: I've never touched this code. You probably know more about it than me now. ;-) ]